### ISSUE TITLE
Increase golint timeout

### ISSUE THREAD

**Issue by roidelapluie** [prometheus/prometheus#8453]
Since the golint update, we see timeouts in the tests:

```
GO111MODULE=on go list -e -compiled -test=true -export=false -deps=true -find=false -tags= -- ./... > /dev/null
GO111MODULE=on /go/bin/golangci-lint run  ./...
ERRO Timeout exceeded: try increasing it by passing --timeout option 
```

We should set a high enough timeout to prevent this.

**Comment by roidelapluie** [comment]
Solved by #8469

**Comment by LeviHarrison** [comment]
The timeout still exceeds quite a bit, for example twice in a row on one of my PRs this morning, so maybe we should increase it further.

https://app.circleci.com/pipelines/github/prometheus/prometheus/12027/workflows/33edc58f-9775-4400-a136-c9844a26545e/jobs/53381 
https://app.circleci.com/pipelines/github/prometheus/prometheus/12026/workflows/9f843f3e-597b-45af-8fe7-b5b6e2b0394f/jobs/53375

**Comment by roidelapluie** [comment]
Now that we have increased it, I think the problem might come from a bug. Golint should run way faster, it might be blocked forever for unknown reasons.

**Comment by LeviHarrison** [comment]
Hmm... I'll investigate.

**Comment by LeviHarrison** [comment]
In theory `--allow-parallel-runners`  should speed things up, but I've found it actually makes it a lot slower. I tried out their Github Action though and after a slow initial run (~2.5 minutes), the subsequent runs were like 30 seconds (https://github.com/LeviHarrison/prometheus/runs/2712125682). You also helpful annotations that point out where the errors are (https://github.com/LeviHarrison/prometheus/commit/1bb34da5a76977710f30b85398ac21708ed7b6f2).
